xapian-core/cluster/cluster.cc

```diff
@@ -81,11 +102,19 @@ TermListGroup::add_document(const Document &document)
 
     TermIterator titer(document.termlist_begin());
 
+    bool allow_stopword_removal = stopper.get() ? true : false;
     for (; titer != document.termlist_end(); ++titer) {
+	const string &term = *titer;
+
+	// Remove stopwords by using the Xapian::Stopper object
+	// that is set
+	if (allow_stopword_removal && (*stopper)(term))
```

Comment:
Oh yes I forgot that these methods get inlined.
I'll wait for your comment on the query raised below to make these changes.